src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/GlobalFlowStateAnalysis/GlobalFlowStateAnalysisValueSet.cs

```diff
@@ -40,17 +40,49 @@ internal sealed class GlobalFlowStateAnalysisValueSet : CacheBasedEquatable<Glob
             Kind = kind;
         }
 
-        public GlobalFlowStateAnalysisValueSet(IAbstractAnalysisValue analysisValue)
-            : this(ImmutableHashSet.Create(analysisValue), ImmutableHashSet<GlobalFlowStateAnalysisValueSet>.Empty, height: 0, GlobalFlowStateAnalysisValueSetKind.Known)
+        public static GlobalFlowStateAnalysisValueSet Create(
+            ImmutableHashSet<IAbstractAnalysisValue> analysisValues,
+            ImmutableHashSet<GlobalFlowStateAnalysisValueSet> parents,
+            int height)
         {
+            Debug.Assert(!analysisValues.IsEmpty || !parents.IsEmpty);
+            return new(analysisValues, parents, height, GlobalFlowStateAnalysisValueSetKind.Known);
         }
 
-        public GlobalFlowStateAnalysisValueSet(GlobalFlowStateAnalysisValueSet parent1, GlobalFlowStateAnalysisValueSet parent2)
-            : this(ImmutableHashSet<IAbstractAnalysisValue>.Empty,
-                   ImmutableHashSet.Create(parent1, parent2),
-                   height: Math.Max(parent1.Height, parent2.Height) + 1,
-                   GlobalFlowStateAnalysisValueSetKind.Known)
+        public static GlobalFlowStateAnalysisValueSet Create(IAbstractAnalysisValue analysisValue)
+            => new(ImmutableHashSet.Create(analysisValue), ImmutableHashSet<GlobalFlowStateAnalysisValueSet>.Empty, height: 0, GlobalFlowStateAnalysisValueSetKind.Known);
+
+        public static GlobalFlowStateAnalysisValueSet CreateWithParents(GlobalFlowStateAnalysisValueSet parent1, GlobalFlowStateAnalysisValueSet parent2)
```

Comment:
💭 It seems like `Merge` would be a better fit for the naming conventions in this library.

Reply:
That would be misleading terminology. `Merge` is only used for standard DFA value set merge at basic blocks with multiple predecessors where incoming data sets from each control flow branch are merged. This is a different operation.